analyzers/src/SonarAnalyzer.CSharp/Rules/VariableShadowsField.cs

```diff
@@ -55,65 +56,90 @@ protected override void Initialize(SonarAnalysisContext context)
 
                     var members = GetMembers(variableSymbol.ContainingType);
 
-                    ReportOnVariableMatchingField(members, declaration.Identifier, c);
+                    ReportOnVariableMatchingField(c, members, declaration.Identifier);
                 },
                 SyntaxKind.ForEachStatement);
 
             context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((LocalDeclarationStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.LocalDeclarationStatement);
+                c => ProcessStatementWithVariableDeclaration(c, (LocalDeclarationStatementSyntax)c.Node, s => s.Declaration), SyntaxKind.LocalDeclarationStatement);
 
             context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((ForStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.ForStatement);
+                c => ProcessStatementWithVariableDeclaration(c, (ForStatementSyntax)c.Node, s => s.Declaration), SyntaxKind.ForStatement);
 
             context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((UsingStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.UsingStatement);
+                c => ProcessStatementWithVariableDeclaration(c, (UsingStatementSyntax)c.Node, s => s.Declaration), SyntaxKind.UsingStatement);
 
             context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((FixedStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.FixedStatement);
+                c => ProcessStatementWithVariableDeclaration(c, (FixedStatementSyntax)c.Node, s => s.Declaration), SyntaxKind.FixedStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c => ProcessStatementWithVariableDesignation(c, (DeclarationPatternSyntaxWrapper)c.Node, s => s.Designation), SyntaxKindEx.DeclarationPattern);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c => ProcessStatementWithVariableDesignation(c, (DeclarationExpressionSyntaxWrapper)c.Node, s => s.Designation), SyntaxKindEx.DeclarationExpression);
+        }
+
+        private static void ProcessStatementWithVariableDesignation<T>(SyntaxNodeAnalysisContext context, T declaration, Func<T, VariableDesignationSyntaxWrapper> variableSelector)
```

Comment:
@martin-strecker-sonarsource  I think this method could be merged with `ProcessSingleVariableDesignation` if one uses the extension methods you implemented, correct?

Reply:
Yes. I used "AllVariables"